### ISSUE TITLE
[Windows] Qt-related issues when building with newly-installed or updated MSYS2 / Mingw64

### ISSUE THREAD

**Issue by Ryunam** [libretro/RetroArch#10414]
## Description

I am getting a "_ZN15QNetworkRequestC1EV" entry point error upon building RA with Mingw64 on Windows 10 x64 with the latest Qt package available (mingw-w64-x86_64-qt5-5.14.2-2), right after launching the executable:

![entrypoint-error](https://user-images.githubusercontent.com/17614150/78570490-585f8700-7825-11ea-9167-2113430eb3a5.png)

This unfortunately prevents the build in question from running: the application does not start at all.

This started to happen after an update I did today of all MSYS2 components and, after a bit of trial and error, I figured out that the culprit was specifically the qt5 package. 
With an older variant (qt5-5.10.1-2, the one that I had installed before updating all packages), the resulting build would run normally.

All redistributable dll's were updated with the redist.7z package for Windows x64 available in the buildbot.

### Environment information

- OS: Windows 10 x64
- Compiler: MSYS2 / Mingw64

**Comment by Ryunam** [comment]
I managed to make RA run by executing these scripts found in the Windows building instructions page (https://docs.libretro.com/development/retroarch/compilation/windows/), which generated new updated libraries:

`for i in $(seq 3); do for bin in $(ntldd -R *exe | grep -i mingw | cut -d">" -f2 | cut -d" " -f2); do cp -vu "$bin" . ; done; done`

`windeployqt --release --no-patchqt --no-translations retroarch.exe
for i in $(seq 3); do for bin in $(ntldd -R imageformats/*dll | grep -i mingw | cut -d">" -f2 | cut -d" " -f2); do cp -vu "$bin" . ; done; done`

However, as a test I backed up these new DLL files that I generated through the script on Mingw64 and replaced them with the ones inside the redist.7z that is currently on the buildbot. As a result, the issue resurfaced.

I am unsure if this is intended / normal behavior, or if there's indeed an actual issue with the redist.7z being outdated.

**Comment by Ryunam** [comment]
Re-opening this, since unfortunately I am still experiencing Qt-related issues.

To recap the whole thing, when building RA for Windows through a newly-installed or freshly updated MSYS2 / Mingw64 environment (as per the instructions described at https://docs.libretro.com/development/retroarch/compilation/windows/) this is what happens:

1) Using the script `"windeployqt --release --no-patchqt --no-translations retroarch.exe"` yields this outcome:
```
Direct dependencies: Qt5Core Qt5Gui Qt5Network Qt5Widgets
All dependencies   : Qt5Core Qt5Gui Qt5Network Qt5Widgets
To be deployed     : Qt5Core Qt5Gui Qt5Network Qt5Widgets
Unable to find the platform plugin.
```

2) As per instructions I can generate new DLL library files by using the following scripts when packaging RA:

`for i in $(seq 3); do for bin in $(ntldd -R *exe | grep -i mingw | cut -d">" -f2 | cut -d" " -f2); do cp -vu "$bin" . ; done; done`

`for i in $(seq 3); do for bin in $(ntldd -R imageformats/*dll | grep -i mingw | cut -d">" -f2 | cut -d" " -f2); do cp -vu "$bin" . ; done; done`

The RA executable launches fine this way. However, when I go to select Show Desktop Menu from within RA, the following error notification appears and leads to a crash:

![Annotation 2020-04-09 180640](https://user-images.githubusercontent.com/17614150/78921465-3bc98600-7a95-11ea-9f59-7764ed6e4d8c.png)

3) If I overwrite the dll libraries with the ones included in the latest redist.7z from the buildbot (https://buildbot.libretro.com/nightly/windows/x86_64/redist.7z) I get this other error:

![entrypoint-error](https://user-images.githubusercontent.com/17614150/78921673-8cd97a00-7a95-11ea-99b6-bb79bd77c406.png)

**Comment by HVR88** [comment]
Confirmed here just today if it matters.  RA can run from the build environment lacking assets, but not from a packaged release

**Comment by Nargash** [comment]
I'm getting something similar now when I compiled from a recent build. When running from a location with the most recent redistributables from the buildbot, I get a message as Ryunam, but a different entry point is missing: _ZNK7QString5splitE5QChar6QFlagsIN2Qt18SplitBehaviorFlagsEENS2_15CaseSensitivityE

Here it is [demangled](http://demangler.com/):
QString::split(QChar, QFlags<Qt::SplitBehaviorFlags>, Qt::CaseSensitivity) const

The same executable launches fine from a directory with no extra libraries.

**Comment by Ryunam** [comment]
I just tested this again. I can confirm that I am getting the same error message that @Nargash is encountering (the entry point is _ZNK7QString5splitE5QChar6QFlagsIN2Qt18SplitBehaviorFlagsEENS2_15CaseSensitivityE) whenever trying to build with a fully-updated MSYS2/mingw64 environment. The qt5 package is once again the culprit here.

Running the script `"windeployqt --release --no-patchqt --no-translations retroarch.exe"` from within the RetroArch folder is still outputting this message as a result:
```
Direct dependencies: Qt5Core Qt5Gui Qt5Network Qt5Widgets
All dependencies   : Qt5Core Qt5Gui Qt5Network Qt5Widgets
To be deployed     : Qt5Core Qt5Gui Qt5Network Qt5Widgets
Unable to find the platform plugin.
```

The only way I've found so far to produce an RA executable that works with the latest redist libraries is by using `pacman -U` from MSYS2 to force-downgrade the qt5 package to 5.10.1-2.

**Comment by Ryunam** [comment]
Yesterday I tried to finally make sense of this issue and I found a reliable way to compile RA correctly with a fully up-to-date MSYS2. This results in a working RA executable with a functional Desktop Menu, while using a qt5 package > 5.10.1-2 (so no need to force-downgrade it).

- Build RA normally (`./configure`, `make clean`, `make -j4`) 
- Copy the executable and all the RA dirs to a new empty folder.
- Generate from scratch the required dll files, by navigating from Mingw to the destination folder using `cd`, then pasting the following scripts: 
1) `for i in $(seq 3); do for bin in $(ntldd -R *exe | grep -i mingw | cut -d">" -f2 | cut -d" " -f2); do cp -vu "$bin" . ; done; done`
2) `for i in $(seq 3); do for bin in $(ntldd -R imageformats/*dll | grep -i mingw | cut -d">" -f2 | cut -d" " -f2); do cp -vu "$bin" . ; done; done`

- Then, from the destination folder, use this command in Mingw:
`windeployqt --no-patchqt --no-translations retroarch.exe`

What made it choke in the first place was the `--release` parameter. According to the Qt issue report:

> Using Qt 5.14.1 with MinGW you cannot provide the -release or -debug flags with windeployqt or it'll puke. You need to let windeployqt self-detect this

Based on this, I think it might be good to update the libretro docs for Windows compilation and remove the --release flag from the windeployqt instruction: https://docs.libretro.com/development/retroarch/compilation/windows/
